wagtailmedia/widgets.py

```diff
@@ -19,6 +29,19 @@ def __init__(self, **kwargs):
         super(AdminMediaChooser, self).__init__(**kwargs)
         self.media_model = get_media_model()
 
+    def get_value_data(self, value):
+        if not WAGTAIL_VERSION > (2, 13):
+            return super().get_value_data(value)
+        if value is None:
+            return value
+        if not isinstance(value, self.media_model):
+            value = self.media_model.objects.get(pk=value)
+        return {
+            "id": value.pk,
+            "title": value.title,
+            "edit_link": reverse("wagtailmedia:edit", args=[value.id]),
+        }
+
     def render_html(self, name, value, attrs):
         instance, value = self.get_instance_and_id(self.media_model, value)
```

Comment:
`get_instance_and_id` is now deprecated so it makes sense to tidy up.

```python
    def render_html(self, name, value, attrs):
        if WAGTAIL_VERSION >= (2, 12):
            # From Wagtail 2.12, get_value_data is called as a preprocessing step in
            # WidgetWithScript before invoking render_html
            value_data = value or {}
        else:
            value_data = self.get_value_data(value) or {}

        original_field_html = super().render_html(name, value_data.get('id'), attrs)

        return render_to_string(
            "wagtailmedia/widgets/media_chooser.html",
            {
                "widget": self,
                "original_field_html": original_field_html,
                "attrs": attrs,
                "value": bool(value_data),  # only used to identify blank values
                "title": value_data.get("title", ""),
                "edit_url": value_data.get("edit_url", ''),
            },
        )
```

and `wagtailmedia/widgets/media_chooser.html`:

```jinja2
{% extends "wagtailadmin/widgets/chooser.html" %}
{% block chooser_class %}media-chooser{% endblock %}

{% block chosen_state_view %}
    <span class="title">{{ title }}</span>
{% endblock %}

{% block edit_chosen_item_url %}{{ edit_url }}{% endblock %}
```

Reply:
Ok, done :).